plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcPageSource.java

```diff
@@ -170,8 +170,7 @@ public Page getNextPage()
 
         completedPositions += page.getPositionCount();
 
-        OptionalLong startRowId = originalFileRowId.isPresent() ?
-                OptionalLong.of(originalFileRowId.get() + recordReader.getFilePosition()) : OptionalLong.empty();
+        OptionalLong startRowId = originalFileRowId.map(rowId -> OptionalLong.of(rowId + recordReader.getFilePosition())).orElseGet(OptionalLong::empty);
```

Comment:
```suggestion
        OptionalLong startRowId = originalFileRowId
                .map(rowId -> OptionalLong.of(rowId + recordReader.getFilePosition()))
                .orElseGet(OptionalLong::empty);
```

is this more readable?

Reply:
Applied.

By the way, it would be nice if we could handle all cleanups of this type (see output of `git grep -n '.isPresent() ?'`)

I will file follow-up PR for this.

Reply:
It might make sense in some places but not everywhere. In some cases the functional code is "unclean" because we end up with a very large lambda to the `map` operation.

It makes sense otherwise.